src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        Figure figure = null;
+        Random random = new Random();
+        int randomNumber = random.nextInt(5);
+        switch (randomNumber) {
+            case 0 :
+                figure = new Circle();
```

Comment:
instead of creating of figure local var and avoid assigning to it, you can directly just return -- new Circle().
If you want to improve your code more, move this object creating to separate methods -- example, return `createCircle()` , etc